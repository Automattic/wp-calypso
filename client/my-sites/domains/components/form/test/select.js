/**
 * @format
 * @jest-environment jsdom
 */

/**
 * External dependencies
 */
import { shallow } from 'enzyme';
import React from 'react';

/**
 * Internal dependencies
 */
import Select from '../select';

describe( '<Select />', () => {
	const defaultProps = {
		label: 'Select label',
		name: 'select',
		onChange: jest.fn(),
		options: [ { label: 'uno', value: 1 }, { label: 'due', value: 2 } ],
		value: '',
		additionalClasses: 'mega-selectzilla',
		errorMessage: null,
	};

	test( 'should render expected output', () => {
		const wrapper = shallow( <Select { ...defaultProps } /> );
		expect( wrapper ).toMatchSnapshot();
	} );

	test( 'should render form validation when there is an error', () => {
		const newProps = {
			...defaultProps,
			errorMessage: 'Duh duh duh!',
		};
		const wrapper = shallow( <Select { ...newProps } /> );
		expect( wrapper.find( 'FormInputValidation' ) ).toHaveLength( 1 );
	} );
} );
