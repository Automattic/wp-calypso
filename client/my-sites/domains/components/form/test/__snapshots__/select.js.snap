// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render expected output 1`] = `
<div
  className="mega-selectzilla select"
>
  <Localized(FormLabel)
    htmlFor="select"
  >
    Select label
  </Localized(FormLabel)>
  <FormSelect
    aria-describedby="validation-field-select"
    disabled={false}
    id="select"
    isError={false}
    name="select"
    onChange={[MockFunction]}
    onClick={[Function]}
    value=""
  >
    <option
      key="1"
      value={1}
    >
      uno
    </option>
    <option
      key="2"
      value={2}
    >
      due
    </option>
  </FormSelect>
</div>
`;
